### PR TITLE
Provide stardoc_output.proto and documentation for proto output

### PR DESCRIPTION
**Pull request by c-parsons**

### CODE REVIEW

File: docs/advanced_stardoc_usage.md [hunk 1]
```diff
@@ -85,3 +86,51 @@ one of the existing canonical [templates](../stardoc/templates/markdown_tables)
...
 
 
+<a name="proto-output"></a>
+## Proto Output
+
+Stardoc provides the option to output documentation information in raw proto
+format. You may find this useful if you need output customization beyond
+Stardoc's current custom-output-template capabilities: you might prefer to build
+your own custom output renderer binary using the data that Stardoc acquires by
+fully evaluating a Starlark file. (If your changes could be incorporated into
```
Comment (jin) [inline docs/advanced_stardoc_usage.md]:
```suggestion
fully evaluating a Starlark file. If your changes could be incorporated into
```

File: stardoc/proto/BUILD [hunk 2]
```diff
@@ -0,0 +1,5 @@
+licenses(["notice"])
+
+package(default_visibility = ["//visibility:public"])
+
+exports_files(glob(["*.proto"]))
```
Comment (jin) [inline stardoc/proto/BUILD]: nit: limit to stardoc_output.proto?

Comment (c-parsons) [reply stardoc/proto/BUILD]: Done!

File: docs/advanced_stardoc_usage.md (hunk 1 above)
Comment (c-parsons) [reply docs/advanced_stardoc_usage.md]: Done.

**Pull request merged**